### PR TITLE
Add support for geth v1.10.18

### LINKED ISSUE #96: New geth version released: Sharblu (v1.10.18)
### What was wrong?<br>
* Support is needed for newly released geth version: `v1.10.18`<br>
### How can it be fixed?<br>
* Add support for geth version `v1.10.18`

### PR DESCRIPTION
**Pull request by fselmo**
### What was wrong?

closes #96

Also... modernizing setup.py because things started breaking (even on `master`). So this will be a breaking PR.

### How was it fixed?

- ran `python update_geth.py v1.10.18`
- update setup.py with versions that shouldn't conflict with `web3.py`
- update tox.ini
- update circleci config

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,14 +17,11 @@
...
-        # Fixing this dependency due to: pytest 3.6.4 has requirement pluggy<0.8,>=0.5, but you'll have pluggy 0.8.0 which is incompatible.
-        "pluggy==0.7.1",
-        # Fixing this dependency due to: requests 2.20.1 has requirement idna<2.8,>=2.5, but you'll have idna 2.8 which is incompatible.
+        "setuptools>=62.3.2",
+        "pluggy>=0.13.1<1",
         "idna==2.7",
-        # idna 2.7 is not supported by requests 2.18
         "requests>=2.20,<3",
-        "tox==2.7.0",
+        "tox>=3.25.0<4",
```
Comment (kclowes) [inline setup.py:24]:
```suggestion
        "tox>=3.25.0,<4",
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -682,6 +683,27 @@ jobs:
         environment:
           GETH_VERSION: v1.10.17
           TOXENV: py39-install-geth-v1.10.17
+  py37-install-geth-v1.10.18:
+    <<: *common_go_v_1_17_1
+    docker:
+      - image: circleci/python:3.7
```
Comment (kclowes) [inline .circleci/config.yml:689]: While we're at it, we might as well update to the `cimg` docker images

File: .circleci/config.yml [hunk 3]
```diff
@@ -682,6 +683,27 @@ jobs:
...
+        environment:
+          GETH_VERSION: v1.10.18
+          TOXENV: py38-install-geth-v1.10.18
+  py39-install-geth-v1.10.18:
+    <<: *common_go_v_1_17_1
+    docker:
+      - image: circleci/python:3.9
+        environment:
+          GETH_VERSION: v1.10.18
+          TOXENV: py39-install-geth-v1.10.18
```
Comment (kclowes) [inline .circleci/config.yml:706]: And probably worth adding a check for Python 3.10?

File: setup.py [hunk 4]
```diff
@@ -17,14 +17,11 @@
     'dev': [
         "bumpversion>=0.5.3,<1",
         "wheel",
-        "setuptools>=36.2.0",
-        # Fixing this dependency due to: pytest 3.6.4 has requirement pluggy<0.8,>=0.5, but you'll have pluggy 0.8.0 which is incompatible.
-        "pluggy==0.7.1",
-        # Fixing this dependency due to: requests 2.20.1 has requirement idna<2.8,>=2.5, but you'll have idna 2.8 which is incompatible.
+        "setuptools>=62.3.2",
+        "pluggy>=0.13.1<1",
```
Comment (kclowes) [inline setup.py:21]: I think this can go

**Review by kclowes — APPROVED** [review]
Thanks for getting to this! If it's not too much trouble, I think we should also add support for Python 3.10 here. I also left comments about this, but I think a couple of the dependencies can be deleted since the various bugs that were causing them to be pinned got fixed in subsequent versions.

**Comment by fselmo** [comment]
@kclowes good points, might as well loop it all together. For the dependencies, I tried to keep them a bit looser. This should be ready for review again.

**Review by kclowes — APPROVED** [review]
Sweet! Looks good to me! 🚢

**Pull request merged**